core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java

```diff
@@ -836,7 +834,7 @@ public synchronized void setIdleTime(long time) {
   }
 
   void startCheckerThread() {
-    checkThreadFactory.get();
+    checkThreadFactory.get().start();
```

Comment:
> If start is called here, and this method is only called once, I'm not sure there's any point in having it memoized anymore.

I agree this could be changed further. This method is only called from ClientContext.getTransportPool, and it's only called once.

Reply:
Additional cleanup in #2612